lib/models/movie.dart

```diff
@@ -0,0 +1,62 @@
+class Movie {
+  final String title;
+  final String originalTitle;
+  final String originalLanguage;
+  final String releaseDate;
+  final num voteAverage;
+  final int voteCount;
+  final List<int> genres;
+  final String overview;
+  final String backdrop;
+  final bool rated;
+  static const baseUrl = 'https://image.tmdb.org/t/p/w500';
+
+  Movie({
+    required this.title,
+    required this.originalTitle,
+    required this.originalLanguage,
+    required this.releaseDate,
+    required this.voteAverage,
+    required this.voteCount,
+    required this.genres,
+    required this.overview,
+    required this.backdrop,
+    required this.rated,
+  });
+
+  factory Movie.mockMovie() {
+    return Movie(
+      title: 'The Super Mario Bros. Movie',
+      originalTitle: 'The Super Mario Bros. Movie',
+      originalLanguage: 'English',
+      releaseDate: '2023-04-05',
+      voteAverage: 7.8,
+      voteCount: 2586,
+      genres: [1],
+      overview:
+          'While working underground to fix a water main, Brooklyn plumbers and brothers Mario and Luigi are transported down a mysterious pipe and wander into a magical new world. But when the brothers are separated, Mario embarks on an epic quest to find Luigi.',
+      backdrop: 'assets/images/mariobros2.jpg',
+      rated: false,
+    );
+  }
+
+  factory Movie.fromJson(json) {
+    return Movie(
+      title: json['title'],
+      originalTitle: json['original_title'],
+      originalLanguage: json['original_language'],
+      releaseDate: json['release_date'],
+      voteAverage: json['vote_average'],
+      voteCount: json['vote_count'],
+      genres: json['genre_ids'].cast<int>(),
+      overview: json['overview'],
+      backdrop: getFullBackdrop(json['backdrop_path']),
+      rated: json['adult'],
+    );
+  }
+
+  static String getFullBackdrop(backdrop){
+    return '$baseUrl$backdrop';
+  }
```

Comment:
Here the idea is to get the backdrop_path and mapping directly to the Movie object like:
`backgrop: json['backdrop_path]`
When you use the backdrop to show the image on the ui side, instead to call `movie.backdrop` you can call `movie.getFullBackdrop`
Whit this change you can change that static method and made a method like this:
`String get fullBackdrop => $baseUrl$backdrop;`

Reply:
done